pkg/publicapi/endpoint/auth/endpoint.go

```diff
@@ -0,0 +1,64 @@
+package auth
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/bacalhau-project/bacalhau/pkg/authn"
+	"github.com/bacalhau-project/bacalhau/pkg/lib/provider"
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/middleware"
+	"github.com/labstack/echo/v4"
+	"github.com/samber/lo"
+)
+
+type Endpoint struct {
+	router   *echo.Echo
+	provider provider.Provider[authn.Authenticator]
+}
+
+func NewEndpoint(ctx context.Context, router *echo.Echo, provider provider.Provider[authn.Authenticator]) *Endpoint {
+	e := &Endpoint{
+		router:   router,
+		provider: provider,
+	}
+
+	g := e.router.Group("/api/v1/auth")
+	g.Use(middleware.SetContentType(echo.MIMEApplicationJSON))
+	g.GET("", e.list)
+
+	for _, key := range provider.Keys(ctx) {
+		method := lo.Must(provider.Get(ctx, key))
+		adaptAuthenticator(method, g.Group("/"+key))
+	}
```

Comment:
Please add a comment clarifying what `keys` are. Similarly what is the return method from `get`? Is `method` the `Authenticator` interface?